plugins/colorbutton/plugin.js

```diff
@@ -503,22 +516,29 @@
 			 * @returns {String} Returns color in hex format, but without the hash at the beginning, e.g. `ff0000` for red.
 			 */
 			normalizeColor: function( color ) {
-				var alphaRegex = /^(rgb|hsl)a\(/g,
-					transparentRegex = /^rgba\((\s*0\s*,?){4}\)$/g,
-					isAlphaColor = alphaRegex.test( color ),
-					// Browsers tend to represent transparent color as rgba(0, 0, 0, 0), so we need to filter out this value.
-					isTransparentColor = transparentRegex.test( color ),
-					colorInstance;
-
-				// For colors with alpha channel we need to use CKEDITOR.tools.color normalization (#4351).
-				if ( isAlphaColor && !isTransparentColor ) {
-					colorInstance = new CKEDITOR.tools.color( color );
-
-					return CKEDITOR.tools.normalizeHex( colorInstance.getHex() || '' ).replace( /#/g, '' );
+				var colorInstance = ( color instanceof CKEDITOR.tools.color ) ? color : new CKEDITOR.tools.color( color, '' ),
```

Comment:
API comment for this method defines `color` as a string. But now, we are verifying if it could be an instance of our `color` class. We should change API comments for that.